src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+            Set<String> microarrayTypes = queryBiomart(getMicroarrayTypesQuery(biomartSpeciesName), 0);
+            // Iterate through each microarray type and retrieve Ensembl-Microarray identifier mappings.
+            // All mappings are stored in a single file, (eg: hsapiens_microarray);
+            retrieveAndStoreMicroarrayData(biomartSpeciesName, getBiomartXMLPath(), microarrayTypes);
+
+            // Query Ensembl-Uniprot (swissprot and trembl) identifier mapping data from Biomart
+            // and write it to a file (eg: hsapiens_uniprot).
+            logger.info("Retrieving UniProt data");
+            retrieveAndStoreUniprotData(biomartSpeciesName, getBiomartXMLPath());
+
+            logger.info("Completed Biomart data retrieval for " + biomartSpeciesName);
+        }
+    }
+
+    /**
+     * Query Biomart for both Uniprot mapping types (curated -- SwissProt, generated -- TrEMBL).
+     * @param biomartSpeciesName
+     * @throws Exception
+     */
+    private void retrieveAndStoreUniprotData(String biomartSpeciesName, String biomartXMLPath) throws IOException {
+        Set<String> uniprotIdLines = new HashSet<>();
+        try {
+            for (String uniprotQueryId : Arrays.asList("uniprotswissprot", "uniprotsptrembl")) {
+                // If proper response, write data to file.
+                Set<String> uniprotResponseLines = queryBiomart(getBiomartIdentifierQuery(biomartXMLPath, biomartSpeciesName, uniprotQueryId), 0);
+                for (String uniprotResponseLine : uniprotResponseLines) {
+                    List<String> tabSplit = Arrays.asList(uniprotResponseLine.split("\t"));
+                    if (tabSplit.size() == 4) {
+                        uniprotIdLines.add(uniprotResponseLine + "\n");
+                    }
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        // Write data to file.
+        if (!uniprotIdLines.isEmpty()) {
+            File speciesUniprotFile = createNewFile(this.destination + biomartSpeciesName + "_uniprot");
+            for (String uniprotIdLine : uniprotIdLines) {
+                Files.write(Paths.get(speciesUniprotFile.toString()), uniprotIdLine.getBytes(), StandardOpenOption.APPEND);
+            }
+        }
+    }
+
+    private String getMicroarrayTypesQuery(String biomartSpeciesName) {
+        return this.getDataURL() + microarrayTypesBaseQuery.replace("BIOMART_SPECIES_NAME", biomartSpeciesName);
+    }
+
+    /**
+     * Query Biomart for Microarray 'identifiers' associated with the microarray type
+     * @param biomartSpeciesName -- String of species name in Biomart format (eg: hsapiens)
+     * @param microarrayTypes -- String, type of microarray identifiers being queried for.
+     * @throws Exception
+     */
+    private void retrieveAndStoreMicroarrayData(String biomartSpeciesName, String biomartXMLPath, Set<String> microarrayTypes) {
```

Comment:
This method looks very similar in logic to `retrieveAndStoreUniProtData`.  Could common elements be extracted to one place (e.g. the test for the tabsplit and the writing of the file contents)?

This could also be broken into two methods to do the retrieval and storage separately like with the UniProt Data.

Reply:
Refactored it to make both microarray/uniprot handled by same functionality